### PR TITLE
Add "conditional rendering" documentation

### PR DESCRIPTION
**Pull request by VladimirZaets**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md [hunk 1]
```diff
@@ -145,3 +145,62 @@ const guestConnection = await register({
...
+                        id: "one_or_less_selected",
+                        label: "Ext one or less selected",
+                    }
+                ]
+            },
+        },
+    },
+});
+```
+The `defaultButtons` constant contains the buttons that will be rendered every time, doesn't meter what condition is.
```
Comment (nitinja) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md:197]: typo found **no matter** what

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md [hunk 2]
```diff
@@ -145,3 +145,62 @@ const guestConnection = await register({
...
+On the line `const selections = await guestConnection?.host?.fragmentSelections?.getSelections?.() || [];` 
+we get information about current list of selected content fragments.
+
+Based on quantity of selected content fragments we rendered different set of buttons.
+
+In case if selected more then one content fragment (`selections.length > 1`) the default buttons and button with id "two_and_more_selected" will be rendered
+In case if selected just one content fragment the default button and button with id "one_or_less_selected" will be rendered
+
+This technic provide possibility to build complex logic behind the buttons rendering.
```
Comment (nitinja) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md:206]: spell suggestion: this technic **provides**

**Review by nitinja — CHANGES_REQUESTED** [review]
I recommend spelling check using google doc or other tool.

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md [hunk 3]
```diff
@@ -145,3 +145,62 @@ const guestConnection = await register({
 | Field    | Type                                                                        | Required | Description                                                                  |
 |----------|-----------------------------------------------------------------------------| ------ |------------------------------------------------------------------------------|
 | id       | `string`                                                                    | ✔️  | "id" is a "data-id" attribute of the button (please use browser developer console to find actual value). |
+
+## Examples:
+### Buttons conditional rendering
+
+In React, rendering occurs whenever the state or props of a component change. If the getButtons method is called within the rendering cycle of a component, it will be executed whenever that component is re-rendered due to state or prop changes.
```
Comment (vkublytskyi) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md:152]: "If the getButtons method is called" - how 3rd party developers know that?
Please rewrite in an assertive way e.g. "getButtons method is called within the rendering cycle of the action bar component."

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md [hunk 4]
```diff
@@ -145,3 +145,62 @@ const guestConnection = await register({
...
+
+You can indeed use this method to dynamically return different sets of buttons based on the data that has changed on the page. For example:
```
Comment (vkublytskyi) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md:154]: Why explain to 3rd party developers internal implementation?
I think you can have only this statement: "Content Fragment Admin invokes getButtons method each time a user changes content fragments selection. Therefore, you can dynamically return different sets of buttons based on the data that has changed on the page. For example:
Comment
"

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md (hunk 1 above)
Comment (vkublytskyi) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md:197]: Please ask other team members if such a detailed code explanation is needed. I think the code is simple enough to be clear for any developer.

**Review by vkublytskyi — CHANGES_REQUESTED** [review]
Please add description of `fragmentSelections?.getSelections?.()` API

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md [hunk 5]
```diff
@@ -145,3 +145,67 @@ const guestConnection = await register({
...
+            },
+        },
+    },
+});
+```
+The `defaultButtons` constant contains the buttons that will be rendered every time, no matter what condition is.
+On the line `const selections = await guestConnection?.host?.fragmentSelections?.getSelections?.() || [];` 
+we get information about current list of selected content fragments.
+
+Based on quantity of selected content fragments we rendered different set of buttons.
```
Comment (irenelagno) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md:206]: Based on the quantity of selected content fragments, we render different sets of buttons.

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md [hunk 6]
```diff
@@ -145,3 +145,67 @@ const guestConnection = await register({
...
+
+In case if selected more then one content fragment (`selections.length > 1`) the default buttons and button with id "two_and_more_selected" will be rendered
+In case if selected just one content fragment the default button and button with id "one_or_less_selected" will be rendered
+
+This technic provides possibility to build complex logic behind the buttons rendering.
```
Comment (irenelagno) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md:211]: Typo(suggestion to rephrase) : This technique allows for building complex logic behind the button rendering.

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md [hunk 7]
```diff
@@ -145,3 +145,67 @@ const guestConnection = await register({
...
+
+### Get List of selected Fragments
+
+You can access data about the currently selected Content Fragments using the `host.fragmentSelections.getSelections()` method in an extension.
+
+
+## Examples:
+### Buttons conditional rendering
+
+Content Fragment Admin invokes getButtons method each time a user changes content fragments selection. Therefore, you can dynamically return different sets of buttons based on the data that has changed on the page.
```
Comment (irenelagno) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md:157]: Content Fragment Admin invokes the `getButtons` method each time a user changes the content fragments selection. Therefore, you can dynamically return different sets of buttons based on the data that has changed on the page.

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md [hunk 8]
```diff
@@ -145,3 +145,67 @@ const guestConnection = await register({
...
+                        id: "one_or_less_selected",
+                        label: "Ext one or less selected",
+                    }
+                ]
+            },
+        },
+    },
+});
+```
+The `defaultButtons` constant contains the buttons that will be rendered every time, no matter what condition is.
```
Comment (irenelagno) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md:202]: The `defaultButtons` constant contains the buttons that will be rendered every time, regardless of the condition.
On the line `const selections = await guestConnection?.host?.fragmentSelections?.getSelections?.() || [];` 
we get information about the current list of selected content fragments.

**Review by irenelagno — CHANGES_REQUESTED** [review]

File: src/pages/services/aem-cf-console-admin/api/action-bar/index.md [hunk 9]
```diff
@@ -145,3 +145,67 @@ const guestConnection = await register({
...
+                        id: "one_or_less_selected",
+                        label: "Ext one or less selected",
+                    }
+                ]
+            },
+        },
+    },
+});
+```
+he defaultButtons constant contains the buttons that will be rendered every time, regardless of the condition.
```
Comment (irenelagno) [inline src/pages/services/aem-cf-console-admin/api/action-bar/index.md]: **The** default..

**Review by irenelagno — APPROVED** [review]

**Pull request merged**